docs/startup/preview/index.md

```diff
@@ -0,0 +1,530 @@
+{% endhighlight %}
+
+### MessageStream
+A MessageStream, as the name implies, represents a stream of messages. A StreamApplication is described as a series of transformations on MessageStreams. You can get a MessageStream in two ways:
+
+1. Using StreamGraph.getInputStream to get the MessageStream for a given input stream (e.g., a Kafka topic).
+
+2. By transforming an existing MessageStream using operations like map, filter, window, join etc.
+
+## Anatomy of a typical Samza StreamApplication
+There are 3 simple steps to write your stream processing applications using the Samza high-level API.
+
+### Step 1: Obtain the input streams:
+You can obtain the MessageStream for your input stream ID (“page-views”) using StreamGraph.getInputStream.
+    {% highlight java %}
+    MessageStream<PageView> pageViewInput = graph.getInputStream(“page-views”, (k,v) -> v);
+    {% endhighlight %}
+
+The first parameter `page-views` is the logical stream ID. Each stream ID is associated with a physical name and a system. By default, Samza uses the stream ID as the physical stream name and accesses the stream on the default system which is specified with the property “job.default.system”. However, the physical name and system properties can be overridden in configuration. For example, the following configuration defines page-views as an alias for the PageViewEvent topic in a local Kafka cluster.
+
+{% highlight jproperties %}
+streams.page-views.samza.system=kafka
+systems.kafka.samza.factory=org.apache.samza.system.kafka.KafkaSystemFactory
+systems.kafka.consumer.zookeeper.connect=localhost:2181
+systems.kafka.producer.bootstrap.servers=localhost:9092
+streams.page-views.samza.physical.name=PageViewEvent
+{% endhighlight %}
+
+The second parameter `(k,v) -> v` is the MessageBuilder function that is used to construct a message from the incoming key and value.
+
+### Step 2: Define your transformation logic:
+You are now ready to define your StreamApplication logic as a series of transformations on MessageStreams.
+
+{% highlight java %}
+MessageStream<DecoratedPageViews> decoratedPageViews
+                                   = pageViewInput.filter(this::isValidPageView)
+                                                  .map(this::addProfileInformation);
+{% endhighlight %}
+
+### Step 3: Write the output to an output stream:
+Finally, you can create an OutputStream using StreamGraph.getOutputStream and send the transformed messages through it.
+
+{% highlight java %}
+// Send messages with userId as the key to “decorated-page-views”.
+decoratedPageViews.sendTo(
+                          graph.getOutputStream(“decorated-page-views”,
+                                                dpv -> dpv.getUserId(),
+                                                dpv -> dpv));
+{% endhighlight %}
+
+The first parameter “decorated-page-views” is a logical stream ID. The properties for this stream ID can be overridden just like the stream IDs for input streams. For example,
```

Comment:
The section on input streams above covers the defaults. Rather than repeating that info here, we refer to the input streams
"can be overridden just like the stream IDs for input streams"